java/internal/src/cuvs_java.c

```diff
@@ -296,20 +290,22 @@ void search_brute_force_index(cuvsBruteForceIndex_t index, float *queries, int t
   int64_t distances_shape[2] = {n_queries, topk};
   DLManagedTensor distances_tensor = prepare_tensor(distances, distances_shape, kDLFloat, 32, 2, kDLCUDA);
 
-  // unpack the incoming long into two 32bit ints
-  for (long i = 0; i < prefilter_data_length; i++) {
-    *(prefilter_data_32 + (2 * i)) = (int)(*(prefilter_data + i) >> 32);
-    *(prefilter_data_32 + ((2 * i) + 1)) = (int)*(prefilter_data + i);
-    //long l = (((long)*(prefilter_data_32 + (2 * i))) << 32) | (*(prefilter_data_32 + ((2 * i) + 1)) & 0xffffffffL);
-  }
-
   cuvsFilter prefilter;
   if (prefilter_data == NULL) {
     prefilter.type = NO_FILTER;
     prefilter.addr = (uintptr_t)NULL;
   } else {
-    int64_t prefilter_shape[1] = {(n_queries * n_rows + 31) / 32};
-    DLManagedTensor prefilter_tensor = prepare_tensor(prefilter_data_d, prefilter_shape, kDLUInt, 32, 1, kDLCUDA);
+    // Parse the filters data
+    uint32_t *prefilters = (uint32_t*) malloc(prefilter_data_length);
```

Comment:
I don't understand why it is necessary to malloc and copy the pre filter data. Is it not possible to access it directly from the passed pointer?

Reply:
I had to change the byte ordering to make it work. Just type casting it to `uint32_t *` didn't work. I'll explore if doing some in-place transformation without allocating a new array is feasible.

Reply:
It should be possible to write the expected byte order at the java level,

Reply:
On the C++ side I added last month an option to specify the byte order used at the time of the creation of the prefilter data: https://github.com/rapidsai/raft/blob/branch-25.04/cpp/include/raft/core/bitmap.hpp#L64.
I think in your case it was created with `uint8_t`? So specifying `original_nbits = sizeof(uint8_t)` should prevent you from that extra allocation/copy. It is not present yet in the C API though.

Reply:
@ChrisHegarty fixed, it was unnecessary to change the order. I was doing things wrong. Fixed now.